graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java

```diff
@@ -268,4 +274,41 @@ public String getPathHome() {
     public Duration getRequestTimeout() {
         return requestTimeout;
     }
+
+    @ValidatorMethod
+    @SuppressWarnings("unused")
+    public void validateElasticsearchHomePath() throws ValidationException {
+        if (getPathHome() != null) {
+            final Path homePath = Paths.get(getPathHome());
+            validateElasticsearchPath(homePath);
+        }
+    }
+
+    @ValidatorMethod
+    @SuppressWarnings("unused")
+    public void validateElasticsearchDataPath() throws ValidationException {
+        if (getPathData() != null) {
+            final Path dataPath = Paths.get(getPathData());
+            validateElasticsearchPath(dataPath);
+        }
+    }
+
+    private void validateElasticsearchPath(@NotNull Path path) throws ValidationException {
+        final Path parent = path.getParent();
+        if (Files.exists(parent) && !Files.isDirectory(parent)) {
+            throw new ValidationException("Path " + parent.toAbsolutePath() + " is not a directory.");
+        }
+
+        if (Files.exists(parent) && !Files.isReadable(parent)) {
+            throw new ValidationException("Path " + parent.toAbsolutePath() + " cannot be read.");
+        }
+
+        if (Files.exists(path) && !Files.isDirectory(path)) {
+            throw new ValidationException("Path " + path.toAbsolutePath() + " is not a directory.");
+        }
+
+        if (Files.exists(path) && !Files.isReadable(path)) {
```

Comment:
Is there a case where the directory also needs to be writable? If so, we should also test for that.